libslide/src/bignum/utils.rs

```diff
@@ -10,6 +15,55 @@ pub fn recast_vec<T, U>(v: Vec<T>) -> Vec<U> {
     unsafe { Vec::from_raw_parts(p as *mut U, len, cap) }
 }
 
+// note we cannot use the above recast since we have a user defined type with
+// an undefined way of storing its memory
+// prereqs: tryfrom(or from) is implemented for type coercion and default and clone are implemented
+pub fn recast_user_vec<T, U>(v: Vec<T>) -> Result<Vec<U>, &'static str>
+where
+    T: std::clone::Clone,
+    U: std::default::Default + std::clone::Clone + std::convert::TryFrom<T>,
+{
+    let mut res: Vec<U> = vec![U::default(); v.len()];
+    for i in 0..v.len() {
+        let val = match U::try_from(v[i].clone()) {
+            Ok(val) => val,
+            Err(_err) => return Err(ERR_MSG),
+        };
+        res[i] = val;
+    }
+    Ok(res)
```

Comment:
Note: this would be more efficient with unsafe code, but we can change that in a later commit.